### PR TITLE
Shadow DOM docs

### PR DESCRIPTION
**Pull request by victorien-provenzano**
### What does this PR do?
This PR introduces Shadow DOM support with Session Replay.

### Motivation
Release of Shadow DOM support with Session Replay.

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/real_user_monitoring/guide/shadow-dom.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+- link: '/real_user_monitoring/session_replay/'
+  tag: 'Documentation'
+  text: 'Learn about Session Replay'
+---
+
+## Overview
+
+Shadow DOM helps developers build more modern websites by allowing them to incorporate isolated and reusable components into their code. Often used in order to keep a clean code structure and avoid style clashes, Shadow DOM usage has become more prominent in modern web development practices. 
+
+Making sure you're able to effectively visualize all the elements which compose your pages in your replays is a key requirement that Datadog can fulfill. Datadog provides full support of Shadow DOM without any extra configuration required — components which are located inside Shadow DOM are automatically captured by Session Replay. In order to get started and have the benefit of enriched browser recordings, set up [Session Replay collection](https://docs.datadoghq.com/real_user_monitoring/session_replay/).
```
[the remark this answers is not in the document]
Comment (jamiemilstein) [reply content/en/real_user_monitoring/guide/shadow-dom.md:15]: @acorretti just double checking here: it's not full support right? we don't support closed shadow DOM?

[the remark this answers is not in the document]
Comment (acorretti) [reply content/en/real_user_monitoring/guide/shadow-dom.md:15]:
```suggestion
Datadog provides support of open Shadow DOM without any extra configuration required — components which are located inside a shadow root are automatically captured by Session Replay.
```
With these changes it'd be less fluff, more insight.

**Comment by jamiemilstein** [comment]
this looks good now iMO!

File: content/en/real_user_monitoring/guide/shadow-dom.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ThibautGeriz) [reply content/en/real_user_monitoring/guide/shadow-dom.md:15]: > Datadog provides full support of Shadow DOM without any extra configuration

It's not true.

Who is the expected reader of this? A technical person?

As a developer I would like to know what is supported & what is not

[the remark this answers is not in the document]
Comment (jamiemilstein) [reply content/en/real_user_monitoring/guide/shadow-dom.md:15]: the reader is a technical person so yes, i like the idea of listing what is not supported, good idea

**Comment by jamiemilstein** [comment]
i would suggest doing the last section in bullets like below:
**What is not supported**

- Closed Shadow DOM
- Dynamic Shadow DOM
- Change in the dynamic CSS style

**Review by jamiemilstein — APPROVED** [review]

**Comment by maycmlee** [comment]
Created [DOCS-4877](https://datadoghq.atlassian.net/browse/DOCS-4877) for review.

File: content/en/real_user_monitoring/guide/shadow-dom.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+- link: '/real_user_monitoring/session_replay/'
+  tag: 'Documentation'
+  text: 'Learn about Session Replay'
+---
+
+## Overview
+
+Shadow DOM helps developers build more modern websites by allowing them to incorporate isolated and reusable components into their code. Often used in order to keep a clean code structure and avoid style clashes, Shadow DOM usage has become more prominent in modern web development practices. 
+
+Starting with `v4.31.0` of the [RUM Browser SDK][1], Datadog provides support of open Shadow DOM without any extra configuration required — components which are located inside a shadow root are automatically captured by Session Replay.
```
Comment (rtrieu) [inline content/en/real_user_monitoring/guide/shadow-dom.md:15]:
```suggestion
## Setup
Starting with `v4.31.0` of the [RUM Browser SDK][1], Datadog provides support of open Shadow DOM without requiring additional configuration. Components which are located inside a shadow root are automatically captured by Session Replay.
```

File: content/en/real_user_monitoring/guide/shadow-dom.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+**Note**: The open Shadow DOM compatibility has been tested on popular frameworks.
+
+## What is not supported
```
Comment (rtrieu) [inline content/en/real_user_monitoring/guide/shadow-dom.md:19]: Similar to my feedback on the other PR, is it possible for us to instead list the things we do support?

**Review by rtrieu — CHANGES_REQUESTED** [review]
Some small suggestions, but otherwise looks good.

File: content/en/real_user_monitoring/guide/shadow-dom.md (hunk 3 above)
Comment (estherk15) [reply content/en/real_user_monitoring/guide/shadow-dom.md:20]: @victorien-provenzano Are we planning on supporting the other modes? Maybe we could add an alert box at the top to highlight that only the open mode is supported. Or if we are not planning on supporting other modes, change the title to `Open Shadow DOM`

Comment (victorien-provenzano) [reply content/en/real_user_monitoring/guide/shadow-dom.md:24]: @estherk15 thanks for the suggestion, went ahead and added an alert box!

**Pull request merged**